internal/pkg/term/progress/spinner_windows.go

```diff
@@ -0,0 +1,142 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package progress
+
+import (
+	"fmt"
+	"io"
+	"text/tabwriter"
+	"time"
+
+	"github.com/aws/copilot-cli/internal/pkg/term/cursor"
+	"github.com/aws/copilot-cli/internal/pkg/term/log"
+	"github.com/briandowns/spinner"
+)
+
+// Events display settings.
+const (
+	minCellWidth           = 20  // minimum number of characters in a table's cell.
+	tabWidth               = 4   // number of characters in between columns.
+	cellPaddingWidth       = 2   // number of padding characters added by default to a cell.
+	paddingChar            = ' ' // character in between columns.
+	noAdditionalFormatting = 0
+)
+
+// TabRow represents a row in a table where columns are separated with a "\t" character.
+type TabRow string
+
+// startStopper is the interface to interact with the spinner.
+type startStopper interface {
+	Start()
+	Stop()
+}
+
+// mover is the interface to interact with the cursor.
+type mover interface {
+	Up(n int)
+	Down(n int)
+	EraseLine()
+}
+
+type writeFlusher interface {
+	io.Writer
+	Flush() error
+}
+
+// Spinner represents an indicator that an asynchronous operation is taking place.
+//
+// For short operations, less than 4 seconds, display only the spinner with the Start and Stop methods.
+// For longer operations, display intermediate progress events using the Events method.
+type Spinner struct {
+	spin startStopper
+	cur  mover
+
+	pastEvents   []TabRow     // Already written entries.
+	eventsWriter writeFlusher // Writer to pretty format events in a table.
+}
+
+// NewSpinner returns a spinner that outputs to stderr.
+func NewSpinner() *Spinner {
+	s := spinner.New(charset, 125*time.Millisecond, spinner.WithHiddenCursor(true))
+	s.Writer = log.DiagnosticWriter
+	return &Spinner{
+		spin:         s,
+		cur:          cursor.New(),
+		eventsWriter: tabwriter.NewWriter(s.Writer, minCellWidth, tabWidth, cellPaddingWidth, paddingChar, noAdditionalFormatting),
+	}
+}
+
+// Start starts the spinner suffixed with a label.
+func (s *Spinner) Start(label string) {
+	s.suffix(fmt.Sprintf(" %s", label))
+	s.spin.Start()
+}
+
+// Stop stops the spinner and replaces it with a label.
+func (s *Spinner) Stop(label string) {
+	s.finalMSG(fmt.Sprintln(label))
+	s.spin.Stop()
+	// Reset event entries once the spinner stops.
+	s.pastEvents = nil
+}
+
+// Events writes additional information below the spinner while the spinner is still in progress.
+// If there are already existing events under the spinner, it replaces them with the new information.
+//
+// An event is displayed in a table, where columns are separated with the '\t' character.
+func (s *Spinner) Events(events []TabRow) {
```

Comment:
Instead of duplicating the entire spinner files, what do you think of having two files:
`events.go` and `events_windows.go` where only the `(s *Spinner) Events(events []TabRow)` method is in these files, since that's the only method that's overriden for windows.

This way it'd be easier to compare the differences.

Reply:
sounds good!